core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryBuilder.java

```diff
@@ -19,115 +19,319 @@
+    public static final boolean DEFAULT_IGNORE_MALFORMED = false;
 
-    private final String name;
-
-    private String distance;
-
-    private double lat;
-
-    private double lon;
+    private final String fieldName;
+    /** Distance from center to cover. */
+    private double distance;
+    /** Point to use as center. */
+    private GeoPoint center = new GeoPoint(Double.NaN, Double.NaN);
+    /** Algorithm to use for distance computation. */
+    private GeoDistance geoDistance = DEFAULT_GEO_DISTANCE;
+    /** Whether or not to use a bbox for pre-filtering. TODO change to enum? */
+    private String optimizeBbox = DEFAULT_OPTIMIZE_BBOX;
+    /** Whether or not to normalize longitude and latitude values to a standard coordinate system */
+    private boolean coerce = DEFAULT_COERCE;
+    /** Whether or not to accept geo points with invalid latitude or longitude */
+    private boolean ignoreMalformed = DEFAULT_IGNORE_MALFORMED;
 
-    private String geohash;
+    static final GeoDistanceQueryBuilder PROTOTYPE = new GeoDistanceQueryBuilder();
 
-    private GeoDistance geoDistance;
-
-    private String optimizeBbox;
-
-    static final GeoDistanceQueryBuilder PROTOTYPE = new GeoDistanceQueryBuilder(null);
+    /** For serialization purposes only. */
+    private GeoDistanceQueryBuilder() {
+        this.fieldName = null;
+    }
 
-    private Boolean coerce;
+    /**
+     * Construct new GeoDistanceQueryBuilder.
+     * @param fieldName name of indexed geo field to operate distance computation on.
+     * */
+    public GeoDistanceQueryBuilder(String fieldName) {
+        Preconditions.checkArgument(Strings.isEmpty(fieldName) == false);
+        this.fieldName = fieldName;
+    }
 
-    private Boolean ignoreMalformed;
+    /** Name of the field this query is operating on. */
+    public String fieldName() {
+        return this.fieldName;
+    }
 
-    public GeoDistanceQueryBuilder(String name) {
-        this.name = name;
+    /** Sets the center point for the query.
+     * @param point the center of the query
+     **/
+    public GeoDistanceQueryBuilder point(GeoPoint point) {
+        this.center = point;
+        return this;
     }
 
+    /**
+     * Sets the center point of the query.
+     * @param lat latitude of center
+     * @param lon longitude of center
+     * */
     public GeoDistanceQueryBuilder point(double lat, double lon) {
-        this.lat = lat;
-        this.lon = lon;
+        this.center = new GeoPoint(lat, lon);
         return this;
     }
 
+    /** Sets the latitude of the center. */
     public GeoDistanceQueryBuilder lat(double lat) {
-        this.lat = lat;
+        if (this.center == null) {
+            this.center = new GeoPoint(lat, Double.NaN);
+        } else {
+            this.center.resetLat(lat);
+        }
         return this;
     }
 
+    /** Sets the longitude of the center. */
     public GeoDistanceQueryBuilder lon(double lon) {
```

Comment:
Can we remove the lat() and lon() methods? We don't want people setting lat but not lon so it don't makes sense (IMO) to allow them to be set separately